gateway/src/main/java/io/camunda/zeebe/gateway/impl/stream/ClientStreamAdapter.java

```diff
@@ -33,6 +35,22 @@ public ClientStreamAdapter(final ClientStreamer<JobActivationProperties> jobStre
   public void handle(
       final StreamActivatedJobsRequest request,
       final ServerCallStreamObserver<ActivatedJob> responseObserver) {
+    if (request.getType().isEmpty()) {
```

Comment:
:wrench: Soft suggestion, but you could rewrite them as [guard clauses](https://en.wikipedia.org/wiki/Guard_(computer_science)) instead of if/else if/else, e.g.

```suggestion
if (request.getType().isBlank()) {
  handleError(responseObserver, "type", "present", "blank");
  return;
}

if (request.getTimeout() < 1) {
  handleError(responseObserver, "timeout", "greater than zero", Long.toString(request.getTimeout()));
  return;
}

handleInternal(request, responseObserver);
```

Pros of guards is you quickly spot the "invalid" cases. Cons can be increased cyclomatic complexity, but in this case it's pretty much the same.

Since the cases are still pretty minimal, I would say it's up to you, even if I have a slight preference toward guards :upside_down_face:

Reply:
Oh, also, `isBlank` might be more interesting here than `isEmpty` - it will also match strings which are purely whitespaces. But I don't know if we want to enforce this to be honest, so I'd leave that up to your team. Is "     " a valid job type? :smile:

Reply:
And you can use `Long.toString` to convert the timeout instead of `String.format`, although you don't really need it. See the comment below.

Reply:
I also prefer guard clauses but I hesitate to use them with empty returns. Since you also prefer them, let's change it. 👍